Harvest.Core/Models/Result.cs

```diff
@@ -21,7 +27,50 @@ public class Result<T>
     // the generic argument. So adding a bit of type inference and implicit casting makes for a better experience without getting too fancy...
     public static class Result
     {
-        public static ResultError Error(string errorMessage) => new ResultError(errorMessage);
+        public static ResultError Error(string messageTemplate, 
+            LogEventLevel logLevel = LogEventLevel.Error,
+            [CallerFilePath] string callerFilePath = "", 
+            [CallerLineNumber] int callerLineNumber = 0)
+        {
+            using var _ = LogContext.PushProperty("FileName", Path.GetFileName(callerFilePath));
```

Comment:
Not pretty, but the _ variable is disposed after function exits, removing FileName and LineNumber properties from log context.